File: apps/admin_backend/src/modules/identity/entities/security-profile.entity.ts
import {
  Column,
  CreateDateColumn,
  Entity,
  JoinColumn,
  OneToOne,
  PrimaryGeneratedColumn,
  UpdateDateColumn,
} from 'typeorm';
import { User } from './user.entity';
import { TotpSecretTransformer } from '../security/totp-secret.transformer';

@Entity('security_profiles')
export class SecurityProfile {
  @PrimaryGeneratedColumn('uuid')
  id: string;

  @Column({ unique: true })
  user_id: string;

  @OneToOne(() => User, (user) => user.security_profile)
  @JoinColumn({ name: 'user_id' })
  user: User;

  @Column({ nullable: true, select: false })
  pin_hash: string | null;

  @Column({ nullable: true, select: false, transformer: TotpSecretTransformer })
  totp_secret_seed: string | null;

  @Column({ default: false })
  is_totp_enabled: boolean;

  @Column({ default: true })
  is_pin_enabled: boolean;

  @CreateDateColumn()
  created_at: Date;

  @UpdateDateColumn()
  updated_at: Date;
}

File: apps/pos_app/lib/data/models/security_profile_entity.dart
import 'package:floor/floor.dart';

@Entity(tableName: 'security_profiles')
class SecurityProfileEntity {
  @primaryKey
  @ColumnInfo(name: 'user_id')
  final String userId;

  @ColumnInfo(name: 'pin_hash')
  final String? pinHash;

  @ColumnInfo(name: 'totp_secret_seed')
  final String? totpSecretSeed;

  @ColumnInfo(name: 'is_totp_enabled')
  final bool isTotpEnabled;

  @ColumnInfo(name: 'is_pin_enabled')
  final bool isPinEnabled;

  SecurityProfileEntity({
    required this.userId,
    this.pinHash,
    this.totpSecretSeed,
    required this.isTotpEnabled,
    required this.isPinEnabled,
  });
}
